charts/gitops-runtime/templates/hooks/pre-install/validate-usage.yaml

```diff
@@ -0,0 +1,49 @@
+{{- if  not .Values.installer.skipUsageValidation }}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: validate-values-config
+  annotations:
+    helm.sh/hook: pre-install
+    helm.sh/hook-delete-policy: hook-succeeded,before-hook-creation,hook-failed
+    helm.sh/hook-weight: "-10"
+data:
+  values.yaml: |
+{{ .Values | toYaml | indent 4 }}
+
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: validate-usage
+  annotations:
+    helm.sh/hook: pre-install
+    helm.sh/hook-delete-policy: hook-succeeded,before-hook-creation
+spec:
+  backoffLimit: 0
+  ttlSecondsAfterFinished: 300
+  template:
+    spec:
+      serviceAccountName: validate-values-sa
```

Comment:
It's using `validate-values-sa`. It isn't very clear, if we want to share the same SA, let's make sure it has generic name. Also after renaming check how helm upgrade handle renamed resources because I don't remember it.